imap/http_dav_sharing.c

```diff
@@ -352,9 +352,10 @@ static int lookup_notify_collection(const char *userid, mbentry_t **mbentry)
             goto done;
         }
 
-        if (*mbentry) free((*mbentry)->name);
-        else *mbentry = mboxlist_entry_create();
+        mboxlist_entry_free(mbentry);
+        *mbentry = mboxlist_entry_create();
         (*mbentry)->name = xstrdup(notifyname);
+        (*mbentry)->mbtype = MBTYPE_COLLECTION;
     }
```

Comment:
This is a slight behaviour change - we would have retained all the fields in the mbentry before if one was passed in, except for the 'name'.

But then - I checked the callsites, and they ALL make sure it's NULL before this is called, so that behaviour was never used.  You could even just `assert(!*mbentry)` to be really sure that the behaviour stays the same!